### PR TITLE
[ADD] estate: Add QWeb view and security

### PR DESCRIPTION
**Pull request by gerashdo**
**Chapter 15**
- Create kanban view for properties list. `selling_price` and `best_price` are conditionally displayed based on property `state`.
- [Reference link](https://www.odoo.com/documentation/16.0/es/developer/tutorials/getting_started/15_qwebintro.html)

**Restrict access to data**
- Define `manager` and `user`(agent) to manage access based on roles.
- Set access rights for each group:
  - `user` (agent) can view, create and edit `estate.property` objects, and have read-only access to other models.
  - `manager` have full CRUD access to all models except delete properties.
- Create field rules for specific restrictions:
  - User can only view properties assigned to them and properties without `salesperson_id`. 
  - Managers can oversee all properties.
- Add `check_access_rights` and `check_access_rule` when setting a property as sold in `estate_account`, to verify if the user has permissions to update a property.
- Add `company_id` to `estate.property` model and implement rule to allow view properties only if they belong to the user company.
- Hide `settings` menu to `user` (agent) role users.
- [Reference link](https://www.odoo.com/documentation/16.0/es/developer/tutorials/restrict_data_access.html#)

### CODE REVIEW

**Comment by gerashdo** [comment]
@JesusValdez96 Ready for review.

File: estate/models/estate_property.py [hunk 1]
```diff
@@ -46,11 +46,13 @@ class EstateProperty(models.Model):
     )
     property_type_id = fields.Many2one("estate.property.type")
     buyer_id = fields.Many2one("res.partner", copy=False)
-    salesperson_id = fields.Many2one("res.users", string="Salesman", default=lambda self: self.env.user)
+    # salesperson_id = fields.Many2one("res.users", string="Salesman", default=lambda self: self.env.user)
+    salesperson_id = fields.Many2one("res.users", string="Salesman")
```
Comment (JesusValdez96) [inline estate/models/estate_property.py]: Why did you change this line? default value was ok.

File: estate_account/models/estate_property.py [hunk 2]
```diff
@@ -16,7 +18,10 @@ def action_set_property_as_sold(self):
             "invoice_line_ids": invoice_line_vals,
         }
 
+        # self.env["account.move"].check_access_rights("write")
+        # print(" reached ".center(100, '='))
```
Comment (JesusValdez96) [inline estate_account/models/estate_property.py:22]: Use comments only to document code.

**Review by JesusValdez96 — CHANGES_REQUESTED** [review]

File: estate/models/estate_property.py (hunk 1 above)
Comment (gerashdo) [reply estate/models/estate_property.py]: One of the exercises asks for it. But I think I misunderstand it 😅. [Link.](https://www.odoo.com/documentation/16.0/es/developer/tutorials/restrict_data_access.html#security-override:~:text=multi%2Dcompany%20rules.-,Exercise,-Add%20a)

image

File: estate_account/models/estate_property.py (hunk 2 above)
Comment (gerashdo) [reply estate_account/models/estate_property.py:22]: I would like to keep in my training project as a reference to know how to check access rights in code. May I?

**Comment by gerashdo** [comment]
@JesusValdez96 Ready for review.

**Comment by gerashdo** [comment]
@JesusValdez96 Friendly reminder.

File: estate_account/models/estate_property.py (hunk 2 above)
Comment (JesusValdez96) [reply estate_account/models/estate_property.py:22]: oh you can save it in a gist

**Review by JesusValdez96 — APPROVED** [review]
